src/client/client4.ts

```diff
@@ -3300,11 +3300,13 @@ export default class Client4 {
     // This function belongs to the Apps Framework feature.
     // Apps Framework feature is experimental, and this function is susceptible
     // to breaking changes without pushing the major version of this package.
-    executeAppCall = async (call: AppCall) => {
-        call.context.user_agent = 'webapp';
+    executeAppCall = async (call: AppCallRequest, type: AppCallType) => {
+        const callCopy = {...call};
+        callCopy.path = `${callCopy.path}/${type}`;
+        callCopy.context.user_agent = 'webapp';
```

Comment:
0/5 seems better to me

```suggestion
        const callCopy: AppCallRequest = {
            ...call,
            path: `${call.path}/${type}`,
            context: {
                ...call.context,
                user_agent: 'webapp',
            },
        };
```